activerecord/lib/active_record/attribute_methods/dirty.rb

```diff
@@ -135,6 +135,18 @@ def keys_for_partial_write
         changed & self.class.column_names
       end
 
+      def attributes_for_create(attributes)
+        if new_record?
+          attributes.push(*keys_with_default_values)
+        end
+
+        attributes
+      end
+
+      def keys_with_default_values
+        self.class._default_attributes.values_before_type_cast.delete_if { |k, v| v.blank? }.keys
```

Comment:
I don't like reaching into the internals like this. The goal has been to decouple these modules further. I don't think the fix for this belongs in `Dirty`. Let's chat more about it on Friday